contracts/orderValidation/OrderValidatorV1.sol

```diff
@@ -339,30 +341,27 @@ contract OrderValidatorV1 {
         address transferManager,
         uint256 tokenId
     ) internal view returns (uint256 validationCode) {
-        (bool answer, bytes memory data) = collection.staticcall(
-            abi.encodeWithSelector(IERC721.getApproved.selector, tokenId)
-        );
-
-        address approvedAddress;
-
-        if (answer) {
-            approvedAddress = abi.decode(data, (address));
-        }
-
-        bool isApprovedSingle = approvedAddress == transferManager;
+        if ((IERC721(collection).ownerOf(tokenId)) != user) return ERC721_TOKEN_ID_NOT_IN_BALANCE;
 
-        (answer, data) = collection.staticcall(
+        (bool success, bytes memory data) = collection.staticcall(
             abi.encodeWithSelector(IERC721.isApprovedForAll.selector, user, transferManager)
         );
 
         bool isApprovedAll;
-
-        if (answer) {
+        if (success) {
             isApprovedAll = abi.decode(data, (bool));
         }
 
-        if (!isApprovedAll && !isApprovedSingle) return ERC721_NO_APPROVAL_FOR_ALL_OR_TOKEN_ID;
-        if ((IERC721(collection).ownerOf(tokenId)) != user) return ERC721_TOKEN_ID_NOT_IN_BALANCE;
+        if (!isApprovedAll) {
+            (success, data) = collection.staticcall(abi.encodeWithSelector(IERC721.getApproved.selector, tokenId));
+
+            address approvedAddress;
+            if (success) {
+                approvedAddress = abi.decode(data, (address));
+            }
+
+            if (!(approvedAddress == transferManager)) return ERC721_NO_APPROVAL_FOR_ALL_OR_TOKEN_ID;
```

Comment:
`f (approvedAddress != transferManager) return ERC721_NO_APPROVAL_FOR_ALL_OR_TOKEN_ID;`

Reply:
It is actually already adjusted